jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/CaseSLAComplianceTest.java

```diff
@@ -0,0 +1,386 @@
+            
+            activeNodes = caseRuntimeDataService.getActiveNodesForCase(caseId, new QueryContext());                        
+            assertThat(activeNodes).hasSize(2);
+            it = activeNodes.iterator();
+            active = it.next();
+            assertThat(active.getName()).isEqualTo("Hello1");
+            active = it.next();
+            assertThat(active.getName()).isEqualTo("[Dynamic] SLA Violation for case " + caseId);
+
+            caseService.cancelCase(caseId);
+            CaseInstance instance = caseService.getCaseInstance(caseId);
+            Assertions.assertThat(instance.getStatus()).isEqualTo(CaseStatus.CANCELLED.getId());
+            caseId = null;
+        } catch (Exception e) {
+            logger.error("Unexpected error {}", e.getMessage(), e);
+            fail("Unexpected exception " + e.getMessage());
+        } finally {
+            if (caseId != null) {
+                caseService.cancelCase(caseId);
+            }
+        }
+    }
+    
+    @Test
+    public void testStartCaseWithSLASubprocess() {
+        Map<String, OrganizationalEntity> roleAssignments = new HashMap<>();
+        roleAssignments.put("owner", new UserImpl("john"));
+        roleAssignments.put("admin", new UserImpl("mary"));
+
+        Map<String, Object> data = new HashMap<>();
+        data.put("s", "Case with SLA");
+        CaseFileInstance caseFile = caseService.newCaseFileInstance(deploymentUnit.getIdentifier(), USER_TASK_SLA_CASE_P_ID, data, roleAssignments);
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), USER_TASK_SLA_CASE_P_ID, caseFile);
+        assertNotNull(caseId);
+        assertEquals(HR_CASE_ID, caseId);
+        try {
+            CaseInstance cInstance = caseService.getCaseInstance(caseId);
+            assertNotNull(cInstance);
+            assertEquals(deploymentUnit.getIdentifier(), cInstance.getDeploymentId());
+            assertEquals(ProcessInstance.SLA_PENDING, cInstance.getSlaCompliance().intValue());
+            
+            Collection<ProcessInstanceDesc> caseProcessInstances = caseRuntimeDataService.getProcessInstancesForCase(caseId, new QueryContext());
+            assertNotNull(caseProcessInstances);
+            assertEquals(1, caseProcessInstances.size());
+            
+            CountDownListenerFactory.getExisting("slaCompliance").waitTillCompleted();
+            
+            cInstance = caseService.getCaseInstance(caseId);
+            assertNotNull(cInstance);
+            assertEquals(deploymentUnit.getIdentifier(), cInstance.getDeploymentId());
+            assertEquals(ProcessInstance.SLA_VIOLATED, cInstance.getSlaCompliance().intValue());
+            
+            caseProcessInstances = caseRuntimeDataService.getProcessInstancesForCase(caseId, new QueryContext());
+            assertNotNull(caseProcessInstances);
+            assertEquals(2, caseProcessInstances.size());
```

Comment:
I would also check that it is a process instance with processId we wanted to start.

Reply:
we don't have process instances to compare with but I'll add check based on correlation key

Reply:
well I meant to check that there is a subprocess you specified in "new org.jbpm.casemgmt.impl.wih.StartProcessSLAViolationListener(\"DataVerification\")"));. But if it is pointless we can leave it as is.